src/core/automation.py

```
@@ -0,0 +1,350 @@
+                    logger.debug(f"Using Quartz CGEvent for key press: {key.name}")
+                    return
+                elif method == 'applescript' and self._send_key_applescript(key):
+                    logger.debug(f"Using AppleScript for key press: {key.name}")
+                    return
+                elif method == 'pyautogui':
+                    result = self._send_key_pyautogui(key)
+                    if result:
+                        logger.debug(f"Using pyautogui fallback for key press: {key.name}")
+                        return
+                    else:
+                        logger.warning(f"pyautogui method returned False")
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)